rollup.config.mjs

```diff
@@ -73,3 +75,27 @@ export default {
     }),
   ],
 };
+
+/**
+ * If env var `THIRD_PARTY_NOTICES_REVISION` is set to `true`, fetches the
+ * revision of a dependency from the npm registry. Falls back to `N/A` if the
+ * revision cannot be fetched for any reason. This is useful for chromedriver
+ * build, as the specific dependency revisions are required for the build.
+ */
+function getRevision(dependency) {
+  if (process.env.THIRD_PARTY_NOTICES_REVISION !== 'true') return 'N/A';
+  try {
+    // Chromium requires the revision to be specified for any dependency.
+    // Even though the revision is not available in the local npm package info,
+    // it can be fetched from the registry.
+    const npmRegistryUrl = `https://registry.npmjs.org/${dependency.name}/${dependency.version}`;
+    const dependencyRegistryInfo = JSON.parse(
+      execSync(`curl -s ${npmRegistryUrl}`).toString()
```

Comment:
this is not cross-platform. Also, I am not sure if the THIRD_PARTY_NOTICES_REVISION is the right file for it?

Reply:
> this is not cross-platform. 

This script is intended to be used only when bumping chromedriver manually, so I'm not sure if being cross-platform is really required. I'm not aware of the cross-platform tool for fetching url. As long as we require a sync way to fetch it (`rollup-plugin-license` does not support async),we can use package like [`sync-request`](https://www.npmjs.com/package/sync-request). WDYT?

> Also, I am not sure if the THIRD_PARTY_NOTICES_REVISION is the right file for it?

`THIRD_PARTY_NOTICES` seems to be a good place for it. Other suggestions?

Reply:
alternative proposal is to download dependency revisions to a file ahead of the rollup call and read in the template. Not blocking though but if someone needs to roll being on Windows it might require extra steps to install curl probably.

Reply:
another alternative, is to post-process the file with adding revisions.

Reply:
The linked bug mentions that it expects the revision in README.chromium, is the README.chromium updated based on the THIRD_PARTY_NOTICES?

Reply:
> The linked bug mentions that it expects the revision in README.chromium, is the README.chromium updated based on the THIRD_PARTY_NOTICES?

there is another script building README.chromium based on `THIRD_PARTY_NOTICES`. Let me experiment if I can add this logic in that script.

Reply:
ah I see, if that does not work out, so current approach LGTM